src/devices/bus/nes_ctrl/zapper.cpp

```diff
@@ -73,14 +115,21 @@ void nes_zapper_device::device_reset()
 {
 }
 
+void nes_bandaihs_device::device_reset()
+{
+	nes_zapper_device::device_reset();
+	m_latch = 0xff;
+	m_strobe = 0xff;
```

Comment:
Does this device actually respond to a system reset, does the reset line not propagate to the controllers?  You shouldn’t really be resetting this stuff if the device doesn’t actually respond to a system reset directly.

Reply:
Well, this begs the question of why do any of the nes_ctrl devices respond to reset (they pretty much all do)? NES controller port pin outs at top left: https://wiki.nesdev.org/w/index.php?title=Controller_port_pinout. Famicom expansion port: https://wiki.nesdev.org/w/index.php?title=Expansion_port. Those don't look promising for peripherals responding to reset.

Reply:
Yes. I've fixed it now.